tests/Fixer/Comment/HeaderCommentFixerTest.php

```diff
@@ -12,195 +12,411 @@
- * @author Antonio J. García Lagar <aj@garcialagar.es>
+ * abc
  */
-class Foo
-{
-}
-EOH;
 
-        $this->doTest($expected, $input);
-    }
+$c;',
+                '<?php
+$c;',
+            ),
+            array(
+                array(
+                    'header' => 'ghi',
+                    'separate' => 'both',
+                ),
+                '<?php
 
-    public function testFixRemovePreviousHeader()
-    {
-        $expected = <<<'EOH'
-<?php
+/*
+ * ghi
+ */
 
-phpinfo();
-EOH;
+$d;',
+                '<?php
+$d;',
+             ),
+            array(
+                array(
+                    'header' => 'ghi',
+                    'separate' => 'top',
+                ),
+                '<?php
 
-        $input = <<<'EOH'
-<?php
+/*
+ * ghi
+ */
+$d;',
+                '<?php
+$d;',
+            ),
+            array(
+                array(
+                    'header' => 'tmp',
+                    'location' => 'after_declare_strict',
+                ),
+                '<?php
 
 /*
- * This file is part of the PHP CS utility.
- *
- * (c) Fabien Potencier <fabien@symfony.com>
- *
- * This source file is subject to the MIT license that is bundled
- * with this source code in the file LICENSE.
+ * tmp
  */
 
-phpinfo();
-EOH;
+declare(ticks=1);
 
-        $fixer = $this->getFixer();
-        $fixer->configure(array('header' => ''));
+echo 1;',
+                '<?php
+declare(ticks=1);
 
-        $this->doTest($expected, $input, null, $fixer);
+echo 1;',
+            ),
+        );
     }
 
-    public function testFixDoNotTouchFilesWithSeveralOpenTags()
+    /**
+     * @return bool|array
+     */
+    protected function getFixerConfiguration()
     {
-        $input = "<?php\nphpinfo();\n?>\n<?";
-        $this->doTest($input);
+        return null === $this->configuration ? array('header' => '') : $this->configuration;
     }
 
-    public function testFixDoNotTouchFilesNotStartingWithOpenTag()
+    public function testDefaultConfiguration()
     {
-        $input = " <?php\nphpinfo();\n";
-        $this->doTest($input);
+        $fixer = $this->getFixer();
+        $method = new \ReflectionMethod($fixer, 'parseConfiguration');
+        $method->setAccessible(true);
+        $this->assertSame(
+            array(
+                "/*\n * a\n */",
+                HeaderCommentFixer::HEADER_COMMENT,
+                HeaderCommentFixer::HEADER_LOCATION_AFTER_DECLARE_STRICT,
+                HeaderCommentFixer::HEADER_LINE_SEPARATION_BOTH,
+            ),
+            $method->invoke($fixer, array('header' => 'a'))
+        );
     }
 
-    public function testFixDoNotTouchFilesWithInlineHtml()
+    /**
+     * @dataProvider provideMisconfiguration
+     */
+    public function testMisconfiguration($configuration, $exceptionMessage)
     {
-        $input = "<?php\nphpinfo();\n?><hr/>";
-        $this->doTest($input);
+        // support for older PHPUnit versions without 'expectException' and 'expectExceptionMessage'
```

Comment:
`setExpectedException` will be removed on PHPUnit 6 and is not available on version 4 (min. by the project). So I wouldn't start using it now.
The annotation is the more reliable, but cannot be used in combination with a dataProvider if the expected exception (message) is provided by the dataProvider (i,e, dynamic exception testing doesn't work with annotations).
I'll think about it some more.